src/javascript/binary/reality-check.js

```diff
@@ -0,0 +1,125 @@
+RealityCheck = (function ($) {
+    "use strict";
+
+    var reality_check_url = page.url.url_for('user/reality_check');
+
+    function RealityCheck(cookieName, persistentStore, logoutLocation) {
+        var val;
+        
+        this.cookieName = cookieName;
+        this.storage = persistentStore;
+
+        val = parseInt($.cookie(this.cookieName).split(',')[0]);
+        if (!val>0) return;     // no or invalid cookie
+
+        this.logoutLocation = logoutLocation;
+        if (!this.logoutLocation) return; // not logged in?
+
+        this.interval = parseInt(val) * 60 * 1000; // convert minutes to millisec
+
+        val = persistentStore.get('reality_check.basetime');
+        if (!val>0) {           // just logged in
+            val = (new Date).getTime();
+            persistentStore.set('reality_check.basetime', val);
+            persistentStore.set('reality_check.ack', 1);
+        }
+        this.basetime = val;
+
+        return this.setAlarm();
+    };
+
+    RealityCheck.prototype.setAlarm = function () {
+        var that = this;
+        var alrm = this.interval - ((new Date).getTime() - this.basetime) % this.interval;
+
+        if (this.tmout) window.clearTimeout(this.tmout);
+
+        this.tmout = window.setTimeout(function () {
+            that.fire();
+        }, alrm);
+    };
+
+    RealityCheck.prototype.fire = function () {
+        var that = this;
+
+        $.ajax({
+            url: reality_check_url,
+            dataType: 'text',
+            success: function (data) {
+                that.display(data);
+            },
+            error: function (xhr) {
+                if (xhr.status === 404) return; // no MF loginid
+                window.setTimeout(function () {
+                    that.fire();
+                }, 5000)
+            },
+        });
+    };
+
+    RealityCheck.prototype.display = function (data) {
+        var that = this, outer, middle, storage_handler; 
+
+        if (outer = $('#reality-check')) outer.remove();
+
+        outer = $("<div id='reality-check'></div>").css({
+            'display':          'table',
```

Comment:
can we move css out of js? we can create sass file or add to current sass file like this `#reality-check { display:table ... }`